### PR TITLE
chore(TDI-48419) Remove smart nexus and use new rules for snapshot build

### PR DESCRIPTION
**Pull request by acatoire**
Remove smart nexus and use new rules for snapshot build

https://jira.talendforge.org/browse/TDI-48809

### CODE REVIEW

**Review by ypiel-talend — COMMENTED** [review]
Some comments

File: Jenkinsfile [hunk 1]
```diff
@@ -127,6 +128,15 @@ pipeline {
...
             DEPLOY : Build snapshot, deploy it to the Nexus for any branch''')
+        string(
+          name: 'VERSION_QUALIFIER',
+          defaultValue: 'DEFAULT',
+          description: '''
+            For branches, used qualifier to store dependencies in the nexus.
+            DEFAULT is {x.y.z}-SNAPSHOT-{jira-id} with:
+              - {x.y.z} the version read form pom.xml
+              - {jira-id} taken from branch name that has to be in the format: user/TDI-XXXXX_Custom_description
+              example: 1.38.0-SNAPSHOT-TDI-48419''')
```
[the remark this answers is not in the document]
Comment (acatoire) [reply Jenkinsfile:139]: corrected

File: Jenkinsfile [hunk 2]
```diff
@@ -127,6 +128,15 @@ pipeline {
             STANDARD : (default) classical CI
             RELEASE : Build release, deploy to the Nexus for master/maintenance branches
             DEPLOY : Build snapshot, deploy it to the Nexus for any branch''')
+        string(
+          name: 'VERSION_QUALIFIER',
+          defaultValue: 'DEFAULT',
+          description: '''
+            For branches, used qualifier to store dependencies in the nexus.
```
[the remark this answers is not in the document]
Comment (acatoire) [reply Jenkinsfile:135]: updated

**Comment by acatoire** [comment]
cancelled after ne PR created

**Pull request closed without merging**